include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_dispatchers.h

```diff
@@ -58,12 +52,25 @@ template <typename _KernelName, bool __is_ascending, ::std::uint8_t __radix_bits
 sycl::event
 __one_wg(sycl::queue __q, _RngPack&& __pack, ::std::size_t __n)
 {
-    using _KeyT = typename _RngPack::_KeyT;
+    using _KeyT = typename ::std::decay_t<_RngPack>::_KeyT;
+    using _EsimRadixSortKernel =
+        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<__esimd_radix_sort_one_wg<_KernelName>>;
+
+    return __radix_sort_one_wg_submitter<__is_ascending, __radix_bits, __data_per_work_item, __work_group_size, _KeyT,
+                                         _EsimRadixSortKernel>()(__q, __pack, __pack, __n);
+}
+
+template <typename _KernelName, bool __is_ascending, ::std::uint8_t __radix_bits, ::std::uint16_t __data_per_work_item,
+          ::std::uint16_t __work_group_size, typename _RngPack1, typename _RngPack2>
+sycl::event
+__one_wg(sycl::queue __q, _RngPack1&& __pack_in, _RngPack2&& __pack_out, ::std::size_t __n)
+{
+    using _KeyT = typename ::std::decay_t<_RngPack1>::_KeyT;
     using _EsimRadixSortKernel =
         oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<__esimd_radix_sort_one_wg<_KernelName>>;
 
     return __radix_sort_one_wg_submitter<__is_ascending, __radix_bits, __data_per_work_item, __work_group_size, _KeyT,
-                                         _EsimRadixSortKernel>()(__q, ::std::forward<_RngPack>(__pack), __n);
+                                         _EsimRadixSortKernel>()(__q, __pack_in, __pack_out, __n);
```

Comment:
Why you not use `std::forward` for pass `__pack_in` and `__pack_out` ?

Reply:
fixed, thanks.